File: Server/TAFRealtimeData.cpp
#include "StdAfx.h"
#include "TAFRealtimeData.h"

TAFRealtimeData::TAFRealtimeData(std::string ticker, boost::posix_time::ptime updatedate) 
: TAFMarketData(ticker, updatedate)
{
	has_last = false;
	has_high = false;
	has_low = false;
	has_volume = false;
	has_open = false;
	has_change = false;
	has_bid = false;
	has_ask = false;
}

std::string TAFRealtimeData::pack()
{
	return ((TAFMarketData)*this).pack() + "&R!" +
		(has_last ? boost::lexical_cast<std::string, double>(last) : "") + "!" +
		(has_high ? boost::lexical_cast<std::string, double>(high) : "") + "!" +
		(has_low ? boost::lexical_cast<std::string, double>(low) : "") + "!" +
		(has_volume ? boost::lexical_cast<std::string, double>(volume) : "") + "!" +
		(has_open ? boost::lexical_cast<std::string, double>(open) : "") + "!" +
		(has_change ? boost::lexical_cast<std::string, double>(change) : "") + "!" +
		(has_bid ? boost::lexical_cast<std::string, double>(bid) : "") + "!" +
		(has_ask ? boost::lexical_cast<std::string, double>(ask) : "");
}

bool TAFRealtimeData::operator==(TAFRealtimeData &d)
{
	if (d.has_last == has_last && d.last == last && d.has_change == has_change && d.has_high == has_high && 
		d.has_low == has_low && d.has_open == has_open && d.has_volume == has_volume && d.has_bid == has_bid && d.has_ask == has_ask
		&& d.change == change && d.high == high && d.low == low && d.open == open && d.volume == volume && d.bid == bid && d.ask == ask)
		return (TAFMarketData)*this == (TAFMarketData)d;
	else
		return false;
}

bool TAFRealtimeData::operator!=(TAFRealtimeData &d)
{
	return !(*this == d);
}

TAFRealtimeData::~TAFRealtimeData(void)
{
}

File: API/TAFRealtimeData.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Meow.FR.TAF.API
{
    public class TAFRealtimeData : TAFMarketData
    {
        public Nullable<double> Open;
        public Nullable<double> High;
        public Nullable<double> Low;
        public Nullable<double> Volume;
        public Nullable<double> Last;
        public Nullable<double> Bid;
        public Nullable<double> Ask;
        public Nullable<double> Change;

        public TAFRealtimeData(string ticker, DateTime updateDate, Nullable<double> last, Nullable<double> high,
            Nullable<double> low, Nullable<double> volume, Nullable<double> open, Nullable<double> change, 
            Nullable<double> bid, Nullable<double> ask)
            : base(ticker, updateDate)
        {
            Last = last;
            High = high;
            Low = low;
            Volume = volume;
            Open = open;
            Change = change;
            Bid = bid;
            Ask = ask;
        }

        public override string ToString()
        {
            return Ticker + " (" + UpdateDate.ToShortDateString() + "): " + Last.ToString();
        }
    }
}
